src/api/app.py

```diff
@@ -140,49 +153,102 @@ def session_id(api_key: str) -> dict[str, int]:
-    """
-    is_active = ...
 
-    return {"is_active": is_active}
-
-
-@get("/close_session", guards=[valid_key_guard], sync_to_thread=False)
-def close_session(api_key: str, session_id: int) -> dict[str, bool]:
+    _session_id = generate_uuid4_int()
+    engine = request.app.state.engine
+    with engine.connect() as conn:
+        conn.execute(
+            sa.text(
+                "INSERT INTO demo_sessions (session_id, api_key, active, start_time, end_time, fake_ip, map, steam_api_data, ingested, created_at, updated_at) VALUES (:session_id, :api_key, :active, :start_time, :end_time, :fake_ip, :map, :steam_api_data, :ingested, :created_at, :updated_at);"  # noqa
+            ),
+            {
+                "session_id": _session_id,
+                "api_key": api_key,
+                "active": True,
+                "start_time": datetime.now().astimezone(timezone.utc).isoformat(),
+                "end_time": None,
+                "fake_ip": fake_ip,
+                "map": map,
+                "steam_api_data": None,
+                "ingested": False,
+                "created_at": datetime.now().astimezone(timezone.utc).isoformat(),
+                "updated_at": datetime.now().astimezone(timezone.utc).isoformat(),
+            },
+        )
+        conn.commit()
+    return {"session_id": _session_id}
+
+
+@get("/close_session", guards=[valid_key_guard, user_not_in_session_guard], sync_to_thread=False)
+def close_session(request: Request, api_key: str, session_id: str) -> dict[str, bool]:
     """Close a session out.
 
     Returns:
         {"closed_successfully": True or False}
     """
-    try:
-        demo_manager.close(session_id)
-    except Exception:
-        ...
-
-    return {"closed_successfully": ...}
-
-
-@websocket_listener("/demos")
-async def demo_session(data: dict[str, str | bytes | int]) -> dict[str, str]:
-    """Handle incoming data from a client upload.
-
-    Smart enough to know where to write data to based on the session ID
-    as to handle reconnecting/duplicates.
-
-    Might want to implement something like
-    https://docs.litestar.dev/2/usage/websockets.html#class-based-websocket-handling
-    because it looks cleaner and likely can handle auth/accepting/valid session id better
-    """
-    demo_manager.handle_demo_data(data)
+    engine = request.app.state.engine
+    current_time = datetime.now().astimezone(timezone.utc).isoformat()
+    with engine.connect() as conn:
+        conn.execute(
+            sa.text(
+                "UPDATE demo_sessions SET active = False, end_time = :end_time, updated_at = :updated_at WHERE session_id = :session_id AND api_key = :api_key;"  # noqa
+            ),
+            {
+                "api_key": api_key,
+                "session_id": session_id,
+                "end_time": current_time,
+                "updated_at": current_time,
+            },
+        )
+        conn.commit()
+
+    return {"closed_successfully": True}
```

Comment:
Could be given it's own function or become a method inside of a class since it is being used in at least one other place, might be a bit of a pre-optimization though. I was just thinking a class that handles all the DB updates with configurable methods to generate the sql statement would be cool/useful for this project.

Reply:
yeah I think it does make sense to move those to lib